src/VivViewer.js

```diff
@@ -1,157 +1,117 @@
+ * @param {VivView} views The various VivViews to render.
+ * */
 export default class VivViewer extends PureComponent {
   constructor(props) {
     super(props);
-    const {
-      initialViewState,
-      overview,
-      viewWidth,
-      viewHeight,
-      loader
-    } = this.props;
-    const vivOverview = overview
-      ? new VivViewerOverview({
-          ...overview,
-          viewWidth,
-          viewHeight,
-          loader,
-          margin: overview.margin,
-          overviewScale: overview.scale
-        })
-      : null;
     this.state = {
-      viewState: {
-        detail: { ...initialViewState, id: 'detail' },
-        overview: overview ? vivOverview.getViewState(initialViewState) : null
-      },
-      overview: vivOverview
+      viewStates: {}
     };
+    const { viewStates } = this.state;
+    const { views } = this.props;
+    views.forEach((view, i) => {
+      // eslint-disable-next-line react/destructuring-assignment
+      viewStates[view.id] = view.getViewState(this.props.viewStates[i]);
+    });
     this._onViewStateChange = this._onViewStateChange.bind(this);
     this.layerFilter = this.layerFilter.bind(this);
   }
 
-  // This prevents only the `draw` call of a layer from firing,
-  // but not other layer lifecycle methods.  Nonetheless, it is
-  // still useful.
+  /**
+   * This prevents only the `draw` call of a layer from firing,
+   * but not other layer lifecycle methods.  Nonetheless, it is
+   * still useful.
+   * @param {Layer} layer The layer being updated.
+   * @param {Viewport} viewport The viewport being updated.
+   * @returns {boolean} Whether or not this layer should be drawn in this viewport
+   */
   // eslint-disable-next-line class-methods-use-this
   layerFilter({ layer, viewport }) {
-    if (layer.id.includes(viewport.id)) {
+    if (layer.id.includes(`-${viewport.id}#`)) {
       // Draw the static layer in the overview
       return true;
     }
     return false;
   }
 
+  /**
+   * This updates the viewState as a callback to the viewport changing in DeckGL
+   * (hence the need for storing viewState in state).
+   */
   _onViewStateChange({ viewId, viewState }) {
     // Save the view state and trigger rerender
-    // only for changes to the `detail` view
-    if (viewId === 'detail') {
-      const { overview } = this.state;
-      const newViewState = {};
-      newViewState.detail = viewState;
-      newViewState.overview = overview
-        ? overview.getViewState(viewState)
-        : null;
-      this.setState({ viewState: newViewState });
-    }
+    // const { overview } = this.state;
+    const { views } = this.props;
+    this.setState(prevState => {
+      const newState = { ...prevState };
+      views.forEach(view => {
+        newState.viewStates = {
+          ...newState.viewStates,
+          [view.id]:
+            view.getViewState({ ...viewState, id: viewId }) ||
+            prevState.viewStates[view.id]
+        };
+      });
```

Comment:
Again getting a little lost in the viewStates for the `VivViews` and viewStates in `VivViewer`